File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/FantasyPointsTeam.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Entity
@Table(name = "fantasy_points_team")
@Data
public class FantasyPointsTeam {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "team_id", referencedColumnName = "id")
    private Team team;

    @ManyToOne
    @JoinColumn(name = "race_id", referencedColumnName = "id")
    private Race race;

    @Column(name = "points", nullable = false)
    private int points;

    @ManyToOne
    @JoinColumn(name = "season_id", referencedColumnName = "id")
    private Season season;
}

File: Formula-Bollo/src/shared/models/fantasyPointsTeam.ts
import { Race } from "./race";
import { Season } from "./season";
import { Team } from "./team";

export class FantasyPointsTeam {
  id: number;
  team: Team;
  race: Race;
  points: number;
  season: Season;

  constructor(id: number, team: Team, race: Race, points: number, season: Season) {
    this.id = id;
    this.team = team;
    this.race = race;
    this.points = points;
    this.season = season;
  }
}
